resources-manager/include/AdaptivePruningManager.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "IRequestHandler.hpp"
+#include "IResourceManager.hpp"
+#include "InternalBFTClient.hpp"
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+enum PruningMode { LEGACY, ADAPTIVE };
+
+namespace concord::performance {
+class AdaptivePruningManager {
+ public:
+  AdaptivePruningManager(const std::shared_ptr<concord::performance::IResourceManager> &resourceManager,
+                         const std::chrono::duration<double, std::milli> &interval,
+                         const std::shared_ptr<bftEngine::impl::IInternalBFTClient> &bftClient);
+  virtual ~AdaptivePruningManager();
+
+  void switchMode(PruningMode newMode) {
+    mode = newMode;
+    conditionVar.notify_one();
+  }
+  PruningMode getCurrentMode() { return mode.load(); }
+
+  void youArePrimary() {
+    amIPrimary = true;
+    conditionVar.notify_one();
+  }
+
+  void youAreNotPrimary() { amIPrimary = false; }
+
+  void setResourceManager(const std::shared_ptr<concord::performance::IResourceManager> &resourceManagerNew) {
```

Comment:
how is the update flow going to happen, who deserialize the configuration and  instantiates the correct instance of IResourceManager?

Reply:
I guess ReconfigurationHandler will parse requests config and initiate the proper object. @yontyon correct?